src/main/java/org/cookieandkakao/babting/domain/meeting/service/MeetingEventService.java

```diff
@@ -0,0 +1,246 @@
+                    .getEventList(getKakaoAccessToken(memberId), from, to)
+                    .events()
+                    .stream()
+                    .map(EventGetResponse::time)
+            )
+            .toList();
+
+        // 시간대 정렬 (시작 시간을 기준으로 오름차순 정렬)
+        List<TimeGetResponse> sortedTimes = allTimes.stream()
+            .sorted(Comparator.comparing(time -> LocalDateTime.parse(time.startAt())))
+            .toList();
+
+        // 겹치는 시간 병합
+        List<TimeGetResponse> mergedTimes = mergeOverlappingTimes(sortedTimes);
+
+        // 빈 시간대 계산
+        return calculateAvailableTimes(mergedTimes, from, to);
+    }
+
+    // 겹치는 시간 병합
+    private List<TimeGetResponse> mergeOverlappingTimes(List<TimeGetResponse> times) {
+        List<TimeGetResponse> mergedTimes = new ArrayList<>();
+
+        if (times.isEmpty())
+            return mergedTimes;
+
+        TimeGetResponse currentTime = times.getFirst();
+
+        for (int i = 1; i < times.size(); i++) {
+            TimeGetResponse next = times.get(i);
+            LocalDateTime currentEnd = LocalDateTime.parse(currentTime.endAt());
+            LocalDateTime nextStart = LocalDateTime.parse(next.startAt());
+
+            // 겹치는 시간대라면 병합
+            if (!nextStart.isAfter(currentEnd)) {
+                currentTime = new TimeGetResponse(
+                    currentTime.startAt(),
+                    maxEndTime(currentTime.endAt(), next.endAt()),
+                    currentTime.timeZone(),
+                    currentTime.allDay() || next.allDay()
+                );
+            } else {
+                mergedTimes.add(currentTime);
+                currentTime = next;
+            }
+        }
+
+        mergedTimes.add(currentTime); // 마지막 시간대 추가
+        return mergedTimes;
+    }
+
+    // 두 시간대 중 더 늦은 종료 시간을 반환
+    private String maxEndTime(String end1, String end2) {
+        LocalDateTime e1 = LocalDateTime.parse(end1);
+        LocalDateTime e2 = LocalDateTime.parse(end2);
+        if (e1.isAfter(e2)){
+            return end1;
+        }
+        return end2;
+    }
+
+    // 빈 시간대
+    private List<TimeGetResponse> calculateAvailableTimes(List<TimeGetResponse> mergedTimes, String from, String to) {
+        List<TimeGetResponse> availableTimes = new ArrayList<>();
+        LocalDateTime searchStart = LocalDateTime.parse(from);
+        LocalDateTime searchEnd = LocalDateTime.parse(to);
+
+        // 첫 번째 시간대 이전의 빈 시간 확인
+        // => 검색 시작일 ~ mergedTime 첫번째 일정의 시작시간까지 빈 시간
+        if (searchStart.isBefore(LocalDateTime.parse(mergedTimes.getFirst().startAt()))) {
+            availableTimes.add(new TimeGetResponse(
+                from,
+                mergedTimes.getFirst().startAt(),
+                "UTC",
+                false
+            ));
+        }
+
+        // 두 시간대 사이의 빈 시간 계산
+        for (int i = 0; i < mergedTimes.size() - 1; i++) {
+            LocalDateTime endOfCurrent = LocalDateTime.parse(mergedTimes.get(i).endAt());
```

Comment:
```suggestion
    LocalDateTime endOfCurrent = LocalDateTime.parse(mergedTimes.getFirst().endAt());
        for (int i = 0; i < mergedTimes.size() - 1; i++) {
            endOfCurrent = maxEndTime(endOfCurrent, LocalDateTime.parse(mergedTimes.get(i).endAt()));
```
이렇게 해주면 병합 과정은 굳이 필요없지 않을까요?
221번째 줄에 if (endOfCurrent.isBefore(startOfNext)) 부분에서도 겹치는 시간인지 확인해주고 있으니 굳이 순회 한 번 더 해가면서 mergedTimes를 구하지 않아도 될 것 같아요

Reply:
엇 그렇네요...! 그런데 제 알고리즘은 O(n) + O(m) (n >= m, m = n -  겹치는 시간 수)이고 연우님 알고리즘은 O(n)이라서 막 엄청난 차이는 아니라서 일단 냅둬볼게요!!....